maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java

```diff
@@ -163,31 +163,33 @@ public void setLanguage(String language) {
     }
 
     /**
-     * @return Description with reason of a Mojo deprecation.
+     * @return <code>true</code> if the Mojo is deprecated, <code>false</code> otherwise.
      */
     public String getDeprecated() {
         return deprecated;
     }
 
     /**
-     * @param deprecated Description with reason of a Mojo deprecation.
+     * @param deprecated <code>true</code> to deprecate the Mojo, <code>false</code> otherwise.
      */
     public void setDeprecated(String deprecated) {
         this.deprecated = deprecated;
     }
 
     /**
-     * @return the list of parameters
+     * @return the list of parameters copy. Any change to returned list is NOT reflected on this instance. To add
+     * parameters, use {@link #addParameter(Parameter)} method.
      */
     public List<Parameter> getParameters() {
-        return parameters;
+        return new ArrayList<>(parameters);
```

Comment:
Maybe we should return unmodifiable collections, it will be difficult to detect if someone try to change this.